src/main/java/com/epam/esm/giftcertificates/filter/Chain.java

```diff
@@ -0,0 +1,48 @@
+package com.epam.esm.giftcertificates.filter;
+
+
+import com.epam.esm.integration.sqlrepo.SQLQuery;
+
+import java.util.*;
+import java.util.function.Function;
+
+public class Chain {
+    private final Map<String, String> parameters;
+    private SearchChainFilter chainFilter;
+    private List<Function<SearchChainFilter, SearchChainFilter>> chainProcessorMap;//todo predicate remove
+
+    public Chain(Map<String, String> parameters) {
+        this.parameters = parameters;
+        buildChain(parameters);
+    }
+
+    private void buildChain(Map<String, String> parameters) {
+        chainProcessorMap = new ArrayList<>();
+        chainProcessorMap.add(parameters.get("sort_name") != null ? chainFilter -> new NameSortSearchChainFilter(chainFilter, parameters.get("sort_name")) : null);
+        chainProcessorMap.add(parameters.get("sort_date") != null ? chainFilter -> new DateSortSearchChainFilter(chainFilter, parameters.get("sort_date")) : null);
+        chainProcessorMap.add(parameters.get("description") != null ? chainFilter -> new DescriptionSearchChainFilter(chainFilter, parameters.get("description")) : null);
+        chainProcessorMap.add(parameters.get("gift_name") != null ? chainFilter -> new GiftNameSearchChainFilter(chainFilter, parameters.get("gift_name")) : null);
+        chainProcessorMap.add(parameters.get("tag_name") != null ? chainFilter -> new TagNameSearchChainFilter(chainFilter, parameters.get("tag_name")) : null);
```

Comment:
Let't inverse responsibilities here a little bit.
Add a static factory method to each ChainFilter class, like that
```java
public static createChain(ChainFilter chainFilter, Map parameters) ChainFilter {....}
```
Made constructor of ChainFilters private, so you can create them only via this new method.
Method should return `new YourChainFilter(...)` or `null` based on codition (in your case this condition will be `parameters.get("description") != null` and others.

so here you can write code like this:
```java
 chainProcessorMap.add(YourChainFilter.createChain(chainFilter, parameters);
chainProcessorMap.add(YourAnotherChainFilter.createChain(chainFilter, parameters);
```